avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/AvroSchemaHelper.java

```diff
@@ -196,6 +196,6 @@ protected static String getTypeId(JavaType type) {
         if (type.isPrimitive()) {
             return ClassUtil.wrapperType(type.getRawClass()).getName();
         }
-        return type.toCanonical();
+        return type.getRawClass().getName();
```

Comment:
The `ListSubtypeTest` will have failing cases when a Jackson schema and Apache deserializer are being used together.